lms/djangoapps/mobile_api/utils.py

```diff
@@ -1,31 +1,92 @@
 """
-Tests for video outline API
+Common utility methods and decorators for Mobile APIs.
 """
 
 
+import functools
+from django.http import Http404
+
+from opaque_keys.edx.keys import CourseKey
 from courseware import access
-from student.roles import CourseBetaTesterRole
-from student import auth
+from courseware.courses import get_course_with_access
+from rest_framework import permissions
+from rest_framework.authentication import OAuth2Authentication, SessionAuthentication
 
 
-def mobile_available_when_enrolled(course, user):
+def mobile_course_access(depth=0, verify_enrolled=True):
     """
-    Determines whether a user has access to a course in a mobile context.
-    Checks if the course is marked as mobile_available or the user has extra permissions
-    that gives them access anyway.
-    Does not check if the user is actually enrolled in the course
+    Method decorator for a mobile API endpoint that verifies the user has access to the course in a mobile context.
     """
+    def _decorator(func):
+        """Outer method decorator."""
+        @functools.wraps(func)
```

Comment:
The `functools.wraps` decorator doesn't override the declaration of the argument list.
Per its doc (see `update_wrapper` at https://docs.python.org/3.1/library/functools.html)
and its code (https://github.com/enthought/Python-2.7.3/blob/master/Lib/functools.py#L16), it only overrides the following attributes of the function: `'__module__' (name of the module the function is defined in), '__name__' (function name), '__doc__' (docstring), and '__dict__' (attributes set on the function)`.